File: Alternating_Digit_Sum.java
class Solution {
    public int alternateDigitSum(int n) {
        int sum=0;
        int sign=1;

        // converting the input integer to string

        String s=Integer.toString(n);

        for(int i=0;i<s.length();i++){
            int v=s.charAt(i)-'0';
            sum+=(v*sign);
            sign=sign*-1;
        }
        return sum;
    }
}

File: Alternating_Digit_Sum.cpp
class Solution {
public:
    int alternateDigitSum(int n) {
        int sum=0;
        string s=to_string(n);
        int len=s.size();
        for(int i=0;i<len;i++)
        {
            if(i%2==0)
            {
                sum+=s[i]-'0';
            }
            else
            {
                sum-=s[i]-'0';
            }
        }
        return sum;
    }
};
